client/client-multi/client-c/src/kaa/kaa_logging.c

```diff
@@ -455,17 +516,38 @@ kaa_error_t kaa_logging_handle_server_sync(kaa_log_collector_t *self
         error_code = kaa_platform_message_read(reader, &delivery_error_code, sizeof(int8_t));
         KAA_RETURN_IF_ERR(error_code);
 
+
         if (delivery_result == LOGGING_RESULT_SUCCESS) {
             KAA_LOG_INFO(self->logger, KAA_ERR_NONE, "Log bucket uploaded successfully, id '%u'", bucket_id);
         } else {
             KAA_LOG_WARN(self->logger, KAA_ERR_WRITE_FAILED, "Failed to upload log bucket, id '%u' (delivery error code '%u')", bucket_id, delivery_error_code);
         }
 
-        remove_request(self, bucket_id);
+
+        /* Count logs uploaded */
+
+        ssize_t count = remove_request(self, bucket_id);
+
+        if (count < 0) {
```

Comment:
Please, remove this if-clause, even if the count variable will be 0 (in case if you change return type of remove_request() to size_t), it's not a good idea to interrupt server sync processing.